src/simulator/cosp_rttov_interface_v13.F90

```diff
@@ -0,0 +1,971 @@
+    call rttov_error('error in rttov options' , lalloc = .false.)
+
+    !%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    ! Figure out how many channels we actually want to reconstruct
+    !%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+    ! Handle different radiance reconstruction options
+    if (nchannels_rec < 0) then
+        print*,'The namelist variable "nchannels_rec" is negative, rttov_direct call will fail. Exiting.'
+        errorstatus = errorstatus_fatal
+        call rttov_exit(errorstatus)
+        ! If the number of channels is negative, don't reconstruct radiances at all
+        rttov_config % nchan_out = 0
+        rttov_config % nchannels_rec = 0 ! Avoid nchanprof set to a negative value
+    else if (nchannels_rec == 0) then
+        ! If the number of channels is set to 0 then reconstruct all instrument channels
+        rttov_config % nchan_out     = rttov_config % coefs % coef % fmv_chn
+        rttov_config % nchannels_rec = rttov_config % coefs % coef % fmv_chn ! Avoid nchanprof set to 0
+    else
+        ! Otherwise read the channel list from the file
+        rttov_config % nchan_out     = nchannels_rec
+        rttov_config % nchannels_rec = nchannels_rec
+    endif
+
+
+    !%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    ! Read in channel indices, emissivities, and reflectivities from .csv if file is passed
+    !%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+    
+    allocate(rttov_config % iChannel(rttov_config % nchan_out)) ! There is a need for these variables to be separate somewhere...
+    allocate(rttov_config % iChannel_out(rttov_config % nchan_out))
+    if (Lwavenum_filepath) then
+        allocate(rttov_config % wavenumChannel(rttov_config % nchan_out))
+        open(18,file=wavenum_filepath,access='sequential',form="formatted")
+        do i = 1, rttov_config % nchan_out
+            read(18,*) rttov_config % wavenumChannel(i)
+        end do
+        close(18)
+    end if
+    if (Lchannel_filepath) then
+        allocate(rttov_config % emisChannel(rttov_config % nchan_out))
+        allocate(rttov_config % reflChannel(rttov_config % nchan_out))
+    
+        open(18,file=channel_filepath,access='sequential',form="formatted")
+        do i = 1, rttov_config % nchan_out
+            read(18,*) rttov_config % iChannel(i), rttov_config % emisChannel(i), rttov_config % reflChannel(i)
+        end do
+        close(18)
+        rttov_config % iChannel_out = rttov_config % iChannel
+    else ! If nothing is passed, compute the first "nchan_out" channels. Ignore emissivity and reflectivity for now.
+        rttov_config % iChannel(:) = (/ (i, i = 1, rttov_config % nchan_out) /)
+        rttov_config % iChannel_out = rttov_config % iChannel
+    endif
+    
+    if (verbose) then
+        print*,'rttov_config % nchan_out:         ',rttov_config % nchan_out
```

Comment:
Same to my other comment, please replace `print` and `write` calls with calls to `errorMessage`